cdc/entry/mounter.go

```diff
@@ -399,84 +408,108 @@ func (m *mounterImpl) mountRowKVEntry(tableInfo *model.TableInfo, row *rowKVEntr
 			TableID:     row.PhysicalTableID,
 			IsPartition: tableInfo.GetPartitionInfo() != nil,
 		},
-		Columns:         cols,
-		PreColumns:      preCols,
-		IndexColumns:    tableInfo.IndexColumnsOffset,
-		ApproximateSize: dataSize,
+		Columns:             cols,
+		PreColumns:          preCols,
+		IndexColumns:        tableInfo.IndexColumnsOffset,
+		ApproximateDataSize: dataSize,
 	}, nil
 }
 
 var emptyBytes = make([]byte, 0)
 
-func formatColVal(datum types.Datum, tp byte) (value interface{}, warn string, err error) {
+const (
+	sizeOfEmptyBytes  = int(unsafe.Sizeof(emptyBytes))
+	sizeOfEmptyString = int(unsafe.Sizeof(""))
+	sizeOfEmptyDatum  = int(unsafe.Sizeof(types.Datum{}))
+)
+
+func sizeOfDatumValue(d types.Datum) int {
+	array := [...]types.Datum{d}
+	return int(types.EstimatedMemUsage(array[:], 1)) - sizeOfEmptyDatum
```

Comment:
Is it accurate to put datum into an array, estimate the memory size and minutes `sizeOfEmptyDatum`